### ISSUE TITLE
API return revoked user ids in SignatureResult

### ISSUE THREAD

**Issue by adithyaphilip** [open-keychain/open-keychain#1723]
Using `getUserIds()` on a `SignatureResult` object returned by  `ACTION_DECRYPT_VERIFY` includes revoked userids.

**Comment by dschuermann** [comment]
The user ids must not only indicate if revoked or not, we must also return if a specific user id is confirmed or not, which is required for XMPP and email verification.

@Valodim Do you plan to extend this API during your work on K-9 Mail? K-9 also needs to cross-check the email address of the sender against the email address in the signature key user id and then display a more specific status for this particular user id if revoked/confirmed/unconfirmed.

**Comment by Valodim** [comment]
Yes, probably, although what that's gonna look like is not planned out yet at all.
